include/wlr/util/addon.h

```diff
@@ -0,0 +1,38 @@
+#ifndef WLR_UTIL_ADDON_H
+#define WLR_UTIL_ADDON_H
+
+#include <wayland-server-core.h>
+
+struct wlr_addon_set {
+	struct wl_list addons;
+};
+
+struct wlr_addon;
+
+struct wlr_addon_interface {
+	const char *name;
+	void (*destroy)(struct wlr_addon *addon);
+};
+
+struct wlr_addon {
+	void *marker;
```

Comment:
Maybe we can remove this field and use `impl` instead? Is there a use-case where we'd need to use a marker different from the `impl`?

Reply:
Only if the owner has more than one addon.

Reply:
I'd prefer to wait a bit and see if someone really needs this. We can always add it later.

Note, one could do something like this and get the same behavior as `owner`/`marker`:

```c
struct my_protocol_impl {
    struct wlr_addon_interface addon_impl;
};
```